samples/touchtest/touchtest.cpp

```diff
@@ -0,0 +1,246 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        touchtest.cpp
+// Purpose:     Multitouch sample
+// Author:      Martin Koegler
+// Modified by:
+// Created:     2015-07-29
+// Copyright:   (c) Martin Koegler
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#include "wx/wx.h"
+
+// Define a new application
+class wxMyApp: public wxApp
+{
+public:
+    bool OnInit() override;
+};
+
+wxDECLARE_APP(wxMyApp);
+
+constexpr unsigned int TOUCH_POINTS = 16;
+
+class MyFrame: public wxFrame
+{
+private:
+    struct TouchState
+    {
+        wxTouchSequenceId id;
+        wxPoint last;
+        wxPen pen;
+    };
+    struct TouchState m_TouchPoints[TOUCH_POINTS];
+    wxTouchSequenceId m_MouseId;
+    wxBitmap m_Bitmap;
+
+    int FindIndexOfTouchId(const wxTouchSequenceId& id);
+
+    void DrawStart(const wxTouchSequenceId& id, wxPoint pos);
+    void DrawUpdate(const wxTouchSequenceId& id, wxPoint pos);
+    void DrawEnd(const wxTouchSequenceId& id, wxPoint pos);
+
+public:
+    MyFrame(wxFrame *parent, const wxString& title,
+        const wxPoint& pos, const wxSize& size, const long style);
+
+    void OnTouchBegin(wxMultiTouchEvent& event);
+    void OnTouchMove(wxMultiTouchEvent& event);
+    void OnTouchEnd(wxMultiTouchEvent& event);
+
+    void OnMouseDown(wxMouseEvent& event);
+    void OnMouseMove(wxMouseEvent& event);
+    void OnMouseUp(wxMouseEvent& event);
+
+    void OnPaint(wxPaintEvent& event);
+    void OnSize(wxSizeEvent& event);
+    void OnQuit(wxCommandEvent& event);
+
+DECLARE_EVENT_TABLE()
+};
+
+IMPLEMENT_APP(wxMyApp)
+
+bool wxMyApp::OnInit()
+{
+    if ( !wxApp::OnInit() )
+        return false;
+
+    wxFrame* frame = new MyFrame(nullptr, wxT("Multi-touch Test"), wxDefaultPosition,
+        wxSize(500, 400), wxDEFAULT_FRAME_STYLE | wxHSCROLL | wxVSCROLL);
```

Comment:
Hardcoding size in pixels doesn't work nice in high DPI, the simplest thing to do is
```suggestion
        wxDefaultSize, wxDEFAULT_FRAME_STYLE | wxHSCROLL | wxVSCROLL);
```

Reply:
With `wxDefaultSize`, the frame becomes too small.

Reply:
Then we need to do something like

```cpp
SetClientSize(FromDIP(wxSize(500, 400)));
```

in the ctor (we can't use `FromDIP()` here as the window doesn't exist yet).